File: Demos/TestCrashReport/TestCrashReport/ViewController.m
//
//  ViewController.m
//  TestCrashReport
//
//  Created by Qihe Bian on 4/24/15.
//  Copyright (c) 2015 LeanCloud Inc. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

- (IBAction) triggerCrash {
  /* Trigger a crash */
  
  CFRelease(NULL);
  
}


- (IBAction) triggerExceptionCrash {
  /* Trigger a crash */
  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
    NSArray *array = [NSArray array];
    [array objectAtIndex:23];
  });

}
@end

File: AVOS/RuntimeTestDemo/ViewController.swift
//
//  ViewController.swift
//  RuntimeTestDemo
//
//  Created by zapcannon87 on 18/01/2018.
//  Copyright © 2018 LeanCloud Inc. All rights reserved.
//

import UIKit

class ViewController: UITableViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 2
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell")!
        switch indexPath.row {
        case 0:
            cell.textLabel?.text = "IM Test"
        case 1:
            cell.textLabel?.text = "Live Query Test"
        default:
            fatalError()
        }
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        switch indexPath.row {
        case 0:
            let vc = self.storyboard!.instantiateViewController(withIdentifier: "IMTestViewController")
            self.navigationController?.pushViewController(vc, animated: true)
        case 1:
            let vc = self.storyboard!.instantiateViewController(withIdentifier: "LiveQueryTestViewController")
            self.navigationController?.pushViewController(vc, animated: true)
        default:
            fatalError()
        }
    }
    
}
